src/AdvancedHMC.jl

```diff
@@ -30,7 +30,20 @@ include("utilities.jl")
 # r: momentum variables
 # z: phase point / a pair of θ and r
 
-include("kinetic.jl")
+# TODO Move it back to hamiltonian.jl after the rand interface is updated
+abstract type AbstractKinetic end
+
+struct GaussianKinetic <: AbstractKinetic end
+
+struct RelativisticKinetic{T} <: AbstractKinetic
```

Comment:
Please consider keeping all these types and functions for relativistic HMC in `experimental/relativistic_hmc.jl`, so we have a clear separation of new research-lead code from the stable code.